openstates/de/bills.py

```diff
@@ -1,693 +1,339 @@
+                attrs.update(**self.categorizer.categorize(action))
+                bill.add_action(**attrs)
+
+        for name, doc in vote_documents.items():
+            vote_chamber = "lower" if "house" in name.lower() else "upper"
+            vote_page = self.lxmlize(doc)
+            vote_info = vote_page.xpath(".//div[@id='page_content']/p")[-1]
+            yes_votes = []
+            no_votes = []
+            other_votes = []                        
+            lines = vote_info.text_content().split("\n")
+            for line in lines:
+                if line.strip().startswith("Date"):
+                    date_str = " ".join(line.split()[1:4])
+                    date = datetime.strptime(date_str,"%m/%d/%Y %I:%M %p")
+                    passed = "Passed" in line
+
+                if line.strip().startswith("Vote Type"):
+                    if "voice" in line.lower():
+                        voice_vote = True
+                    else:
+                        voice_vote = False
+                        yes_count = int(re.findall("Yes: (\d*)",line)[0])
+                        no_count = int(re.findall("No: (\d*)",line)[0])
+                        other_count = int(re.findall("Not Voting: (\d*)",line)[0])
+                        other_count += int(re.findall("Absent: (\d*)",line)[0])
+                        vote_tds = vote_page.xpath(".//table//td")
+                        person_seen = False
+                        for td in vote_tds:
+                            if person_seen:
+                                person_vote = td.text_content().strip()
+                                if person_vote == "Y":
+                                    yes_votes.append(person)
+                                elif person_vote == "N":
+                                    no_votes.append(person)
+                                elif person_vote in ["NV","A","X"]:
+                                    other_votes.append(person)
+                                else:
+                                    raise AssertionError("Unknown vote '{}'".format(person_vote))
+                                person_seen = False
+                            else:
+                                person = td.text_content().strip()
+                                if person:
+                                    person_seen = True
+
+            if voice_vote:
+                vote = Vote(vote_chamber,date,"passage",passed,0,0,0)
+            else:
+                vote = Vote(vote_chamber,date,"passage",
+                            passed,yes_count,no_count,other_count,
+                            yes_votes=[],
+                            no_votes=[],
+                            other_votes=[])
+
+                vote["yes_votes"] = yes_votes
+                vote["no_votes"] = no_votes
+                vote["other_votes"] = other_votes
+
+            assert len(vote["yes_votes"]) == vote["yes_count"], \
```

Comment:
didn't we decide these get called when the vote is saved? I'll just delete the asserts.